src/realm/object-store/dictionary.cpp

```diff
@@ -217,7 +217,6 @@ Mixed Dictionary::get_any(StringData key)
 
 Mixed Dictionary::get_any(size_t ndx) const
 {
-    verify_valid_row(ndx);
```

Comment:
I think all these places in collection subclasses where `verify_valid_row(ndx);` is removed should still be calling  `verify_attached()`? It was being called indirectly from `size()` during the index check before.

Reply:
I added `verify_attached()` a few places. In most of the cased there were a call to `verify_in_transaction()` which would also check attachment.